homeassistant/components/vallox/sensor.py

```diff
@@ -264,16 +117,131 @@ class ValloxFilterRemainingSensor(ValloxSensor):
     async def async_update(self):
         """Fetch state from the ventilation unit."""
         try:
-            days_remaining = int(self._state_proxy.fetch_metric(self._metric_key))
+            days_remaining = int(
+                self._state_proxy.fetch_metric(self.entity_description.metric_key)
+            )
             days_remaining_delta = timedelta(days=days_remaining)
 
             # Since only a delta of days is received from the device, fix the
             # time so the timestamp does not change with every update.
             now = datetime.utcnow().replace(hour=13, minute=0, second=0, microsecond=0)
 
-            self._state = (now + days_remaining_delta).isoformat()
-            self._available = True
+            self._attr_native_value = (now + days_remaining_delta).isoformat()
+            self._attr_available = True
 
         except (OSError, KeyError) as err:
-            self._available = False
+            self._attr_available = False
             _LOGGER.error("Error updating sensor: %s", err)
+
+
+@dataclass
+class ValloxOptionalKeysMixin:
```

Comment:
This is only needed for keys without a default, both these should go under `ValloxSensorEntityDescription`

Reply:
Thanks, done.